mypy/sametypes.py

```diff
@@ -123,9 +123,14 @@ def visit_typeddict_type(self, left: TypedDictType) -> bool:
             return False
 
     def visit_union_type(self, left: UnionType) -> bool:
-        # XXX This is a test for syntactic equality, not equivalence
         if isinstance(self.right, UnionType):
-            return is_same_types(left.items, self.right.items)
+            for item in left.items:
```

Comment:
You also need to verify that all items in the right type match some item in the left union. Now `Union[int, str]` is considered the same as `Union[int, str, bytes]`, but not the other way around. Also add a test case for this.